### ISSUE TITLE
Training

### ISSUE THREAD

**Issue by MarijnJABoer** [naturalis/FormicID#16]
## Tuning hyperparameters
- [x] Steps per epoch (`samples//batch_size`) (the same as iterations per epoch? - Yes!?)
- [x] Optimizer (nadam!?)
  - [x] Tune the optimizers
- [x] Epoch (related to how diverse your data is)
- [x] Batch size (because of the low quality images, we can increase this?) [link](https://stats.stackexchange.com/questions/164876/tradeoff-batch-size-vs-number-of-iterations-to-train-a-neural-network)
- [x] Dropout (what percentage?)
- [x] Implement hyperparameter grid search?

## Training
- [x] Load model from `build.py` #8 
- [x] Load model from `models.py` #8 
- [x] `Earlystopping`
- [x] `ReduceLROnPlateau`
- [x] Move `ImageDataGenerator` from data_input.py to train.py
- [x] Split `train_generator` and `train_datagen` in `train.py`
  - [x] Implement this fix in `show_augmentation()`
- [x] Multi-GPU model [link](https://keras.io/utils/#multi_gpu_model)
> *On model saving with Multi-GPU*
> To save the multi-gpu model, use .save(fname) or .save_weights(fname) with the template model (the argument you passed to multi_gpu_model), rather than the model returned by multi_gpu_model.
- [x] Load model using imagenet weights
- [x] Metrics to csv
- [x] Add Inception-ResNet v2
- [x] Building my own image data generator [link](https://github.com/keras-team/keras/issues/1638)
  - [x] `.flow_from_directory()` is faster than `.flow()`
  - [link](https://github.com/aurora95/Keras-FCN/issues/50#issuecomment-367978740)
- [x] Import callbacks correctly
- [x] Pass training to object  (`history = model.fit()`)
 > `model.fit()` returns history object which contains information about training an validation accuracy and loss:
```
{'acc': [0.9843952109499714],
 'loss': [0.050826362343496051],
 'val_acc': [0.98403786838658314],
 'val_loss': [0.0502210383056177]
}
```
- [x] plot metrics using python (with ggplot theme)
- [x] is `categorical_crossentropy` the correct loss function?

## Metrics
- [x] Fix `metrics=['accuracy']`
- [x] Add RMSE
- [x] Add `categorical_accuracy`
- [x] Add more metrics? [link](https://machinelearningmastery.com/custom-metrics-deep-learning-keras-python/)
  - MAE
  - MSE
- [x] Add classification metrics for top 3 and top 5 `top_k_categorical_accuracy`

## Others
- [x] Is resizing / augmentation working?
- [x] `standardization` vs `normalization` vs `generalization` vs `regularization`?
- [x] Choose a verbose 
  - verbose: 0, 1, or 2. Verbosity mode. 0 = silent, 1 = progress bar, 2 = one line per epoch.
- [x] Sessions
```python
    sess = tf.session()
    K.set_session(sess)
```
